File: backend/app/api/snippets.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.models import UserSnippet
from app.database import get_db
router = APIRouter()

@router.get("/api/snippets/{user_id}")
def get_snippets(user_id: int,db:Session = Depends(get_db)):
    snippets = db.query(UserSnippet).filter_by(user_id=user_id).all()
    return snippets

@router.post("/api/snippets")
def add_snippet(snippet: dict,db:Session = Depends(get_db)):
    snippet_obj = UserSnippet(**snippet)
    db.add(snippet_obj)
    db.commit()
    return {"status": "saved"}

File: frontend/src/lib/api/snippets.ts
// src/lib/api/snippets.ts
import api from "../api";

export type UserSnippet = {
  id?: number;
  user_id: number;
  title: string;
  language: string;
  code: string;
  notes?: string;
};

/** Fetch all snippets for a user */
export async function getSnippets(userId: number): Promise<UserSnippet[]> {
  const { data } = await api.get(`/api/snippets/${userId}`);
  return data;
}

/** Save a new snippet */
export async function addSnippet(snippet: UserSnippet): Promise<{ status: string }> {
  const { data } = await api.post("/api/snippets", snippet);
  return data;
}
